app/uk/gov/hmrc/pillar2/service/SubscriptionService.scala

```diff
@@ -491,144 +417,32 @@ class SubscriptionService @Inject() (
-      pContactEmail     <- userAnswers.get(subPrimaryEmailId)
-      sContactNominated <- userAnswers.get(subAddSecondaryContactId)
-      accountingPeriod  <- userAnswers.get(subAccountingPeriodId)
-      nominatedFm       <- userAnswers.get(NominateFilingMemberId)
-      registrationDate  <- userAnswers.get(subRegistrationDateId)
-      plrReference      <- userAnswers.get(plrReferenceId)
-    } yield {
-      val upeDetail = UpeDetailsAmend(
-        plrReference = plrReference,
-        customerIdentification1 = userAnswers.get(upeRegInformationId).map(_.crn),
-        customerIdentification2 = userAnswers.get(upeRegInformationId).map(_.utr),
-        organisationName = companyName,
-        registrationDate = registrationDate,
-        domesticOnly = if (mneOrDom == MneOrDomestic.Uk) true else false,
-        filingMember = nominatedFm
-      )
-      val primaryContact =
-        ContactDetailsType(name = pContactName, telephone = userAnswers.get(subPrimaryCapturePhoneId), emailAddress = pContactEmail)
-      val secondaryContact = if (sContactNominated) {
-        for {
-          name         <- userAnswers.get(subSecondaryContactNameId)
-          emailAddress <- userAnswers.get(subSecondaryEmailId)
-        } yield ContactDetailsType(name = name, telephone = userAnswers.get(subSecondaryCapturePhoneId), emailAddress = emailAddress)
-      } else {
-        None
-      }
-
-      val filingMember = if (nominatedFm) {
-        userAnswers
-          .get(subFilingMemberDetailsId)
-          .map(det =>
-            FilingMemberAmendDetails(
-              safeId = det.safeId,
-              customerIdentification1 = det.customerIdentification1,
-              customerIdentification2 = det.customerIdentification2,
-              organisationName = det.organisationName
+  def sendAmendedData(id: String, amendData: AmendSubscriptionSuccess)(implicit hc: HeaderCarrier): Future[Done] =
+    subscriptionConnector.amendSubscriptionInformation(amendData).flatMap { response =>
+      if (response.status == 200) {
+        auditService.auditAmendSubscription(requestData = amendData, responseData = AuditResponseReceived(response.status, response.json))
+        response.json.validate[AmendResponse] match {
+          case JsSuccess(result, _) =>
+            logger.info(
+              s"Successful response received for amend subscription for form ${result.success.formBundleNumber} at ${result.success.processingDate}"
             )
-          )
+            repository.remove(id)
+            Future.successful(Done)
+          case _ => Future.failed(JsResultError)
+        }
       } else {
-        None
+        logger.info(
+          s"Unsuccessful response received for amend subscription with ${response.status} status and body: ${response.body} "
+        )
+        auditService.auditAmendSubscription(requestData = amendData, responseData = AuditResponseReceived(response.status, response.json))
+        Future.failed(UnexpectedResponse)
```

Comment:
test for UnexpectedResponse ?

Reply:
I already have tests for them, please see [here](https://github.com/hmrc/pillar2/blob/e5b059bc8d6bfae6bebecd9cb566b8b971d1e1fb/test/uk/gov/hmrc/pillar2/services/SubscriptionServiceSpec.scala#L135)